src/Model/Core/BunqModel.php

```diff
@@ -121,17 +135,85 @@ protected static function createListFromResponseArray(
      *
      * @return BunqModel|null
      */
-    protected static function createFromResponseArray(array $responseArray, string $wrapper = null)
-    {
+    protected static function createFromResponseArray(
+        array $responseArray,
+        string $wrapper = null
+    ) {
+        if (is_subclass_of(get_called_class(), AnchorObjectInterface::class)) {
+            return self::createFromResponseArrayAnchorObject($responseArray);
+        }
+
         if (is_string($wrapper)) {
             $responseArray = $responseArray[$wrapper];
         }
 
         if (is_null($responseArray)) {
             return null;
-        } else {
-            return self::createInstanceFromResponseArray($responseArray);
         }
+
+        return self::createInstanceFromResponseArray($responseArray);
+    }
+
+    /**
+     * @param mixed[] $responseArray
+     * @param int|null $depthCounter
+     *
+     * @return BunqModel|null
+     */
+    private static function createFromResponseArrayAnchorObject(
+        array $responseArray,
+        int $depthCounter = null
+    ) {
+        if (is_null($responseArray)) {
+            return null;
+        }
+
+        if (is_null($depthCounter)) {
+            $depthCounter = self::DEPTH_COUNTER_BEGIN;
+        }
+
+        $model = self::createInstanceFromResponseArray($responseArray);
+
+        if ($model->isAllFieldNull() && $depthCounter < self::DEPTH_COUNTER_MAX) {
+            $model = self::decodeInsideModelFields($responseArray, $model, $depthCounter);
+        }
+
+        return $model;
+    }
+
+    /**
+     * @param mixed[] $responseArray
+     * @param BunqModel $model
+     * @param int $depthCounter
+     *
+     * @return BunqModel
+     */
+    private static function decodeInsideModelFields(array $responseArray, BunqModel $model, int $depthCounter)
+    {
+        $modelFields = (array_keys(get_object_vars($model)));
+
+        foreach ($modelFields as $field) {
+            try {
+                $fieldClass = ModelUtil::determineModelClassNameQualified($field);
+                $reflectionClass = new ReflectionClass($fieldClass);
+                /** @var BunqModel $bunqModelSubClass */
+                $bunqModelSubClass = $reflectionClass->newInstanceWithoutConstructor();
+                $fieldContents = $bunqModelSubClass::createFromResponseArrayAnchorObject(
+                    $responseArray,
+                    $depthCounter + self::DEPTH_COUNTER_INCREMENTER
+                );
+
+                if ($fieldContents->isAllFieldNull()) {
+                    $model->{$field} = null;
+                } else {
+                    $model->{$field} = $fieldContents;
+                }
+            } catch (BunqException $exception) {
+                continue;
```

Comment:
mmm this sucks

Reply:
Correct I dont even remember why I did this 😅 let me find out.